<launch>
  <group ns="quadrotor">

    <arg name="mass" value="0.5"/>
    <param name="mass" value="$(arg mass)"/> 

    <rosparam file="$(find mav_manager)/config/params.yaml"/>

    <node pkg="quadrotor_simulator"
      type="quadrotor_simulator_so3"
      name="quadrotor_simulator_so3"
      output="screen">
      <param name="rate/odom" value="100.0"/>
      <remap from="~odom" to="odom"/>
      <remap from="~imu" to="imu"/>
      <remap from="~cmd" to="so3_cmd"/>
    </node>

    <node pkg="nodelet"
      type="nodelet"
      name="standalone_nodelet"
      args="manager"
      output="screen"/>

    <node pkg="nodelet"
      type="nodelet"
      name="trackers_manager"
      args="load trackers_manager/TrackersManager standalone_nodelet"
      output="screen">
      <rosparam file="$(find trackers_manager)/config/trackers.yaml"/>
      <rosparam file="$(find so3_control)/config/gains.yaml"/>
      <remap from="~odom" to="odom"/>
      <remap from="~cmd" to="position_cmd"/>
    </node>

    <node pkg="nodelet"
      type="nodelet"
      args="load so3_control/SO3ControlNodelet standalone_nodelet"
      name="so3_control"
      required="true"
      output="screen">
      <rosparam file="$(find so3_control)/config/gains.yaml"/>
      <param name="mass" value="$(arg mass)"/>
      <remap from="~odom" to="odom"/>
      <remap from="~position_cmd" to="position_cmd"/>
      <remap from="~so3_cmd" to="so3_cmd"/>
      <remap from="~motors" to="motors"/>
    </node>

    <node pkg="mesh_visualization"
      type="mesh_visualization"
      name="mesh_visualization"
      output="screen">
      <param name="mesh_resource" value="package://mesh_visualization/mesh/hummingbird.mesh"/>
      <param name="color/r" value="0.0"/>
      <param name="color/g" value="0.0"/>
      <param name="color/b" value="1.0"/>
      <param name="color/a" value="0.7"/>
      <remap from="~odom" to="odom"/>
    </node>

    <node pkg="mav_manager"
      type="mav_manager_node"
      name="mav_manager_node"
      output="screen">
      <param name="startup_sleep_duration" value="0.25"/>
      <remap from="output_data" to="quad_decode_msg/output_data"/>
    </node>

  </group>
</launch>
